src/plugins/profiling/server/routes/flamegraph.ts

```diff
@@ -93,58 +94,36 @@ export class FlameGraph {
   }
 
   toElastic() {
-    // Create a lookup table for stack frames with an appropriate default if
-    // the corresponding metadata is missing
-    const frameMap = new Map();
-    for (let i = 0; i < this.stackframes.length; i++) {
-      const frame = this.stackframes[i];
-      if (frame.found) {
-        frameMap.set(frame._id, frame._source);
-      } else {
-        frameMap.set(frame._id, {
-          FunctionName: '',
-          LineNumber: 0,
-        });
-      }
-    }
+    const leaves = [];
+    let n = 0;
 
-    // Create a lookup table for executables with an appropriate default if
-    // the corresponding metadata is missing
-    const exeMap = new Map();
-    for (let i = 0; i < this.executables.length; i++) {
-      const exe = this.executables[i];
-      if (exe.found) {
-        exeMap.set(exe._id, exe._source);
-      } else {
-        exeMap.set(exe._id, {
-          FileName: '',
-        });
-      }
-    }
+    for (const trace of this.stacktraces.values()) {
+      const path = ['root'];
+      for (let i = trace.FrameID.length - 1; i >= 0; i--) {
+        const label = getLabel(
+          this.stackframes.get(trace.FrameID[i]),
+          this.executables.get(trace.FileID[i]),
+          parseInt(trace.Type[i], 10)
+        );
 
-    let leaves = [];
-
-    for (const trace of this.stacktraces) {
-      if (trace.found) {
-        const path = ['root'];
-        for (let i = 0; i < trace._source.FrameID.length; i++) {
-          const label = getLabel(
-            frameMap.get(trace._source.FrameID[i]),
-            exeMap.get(trace._source.FileID[i]),
-            trace._source.Type[i]);
-          if (label.length === 0) {
-            path.push(trace._source.FrameID[i]);
-          } else {
-            path.push(label);
-          }
+        if (label.length === 0) {
+          path.push(trace.FrameID[i]);
+        } else {
+          path.push(label);
         }
-        const leaf = {
-          id: path[path.length - 1],
-          value: 1,
-          depth: trace._source.FrameID.length,
-          pathFromRoot: Object.fromEntries(path.map((item, i) => [i, item])),
-        };
-        leaves.push(leaf);
+      }
+      const leaf = {
+        id: path[path.length - 1],
+        value: 1,
+        depth: trace.FrameID.length,
+        pathFromRoot: Object.fromEntries(path.map((item, i) => [i, item])),
+      };
+      leaves.push(leaf);
+
+      n++;
+      if (n >= 1000) {
+        // just don't overload the Kibana flamechart
+        break;
```

Comment:
We should be more explicit about this limit IMO, with a constant or a small wrapper function, but we can refine after this PR

Reply:
This is just temporary to avoid OOMs while testing. With the new Flamechart (hopefully coming soon), we can remove this work-around completely.